### PR TITLE
Add scope_definition examples

### PR DESCRIPTION
**Pull request by LeoLiuYan**

### CODE REVIEW

File: examples/typed-component/sample_scope_definition.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+apiVersion: core.oam.dev/v1alpha2
+kind: ScopeDefinition
```
Comment (wonderflow) [inline examples/typed-component/sample_scope_definition.yaml:2]: please move this to definition.yaml

File: examples/typed-component/sample_scope_definition.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            kind: ManualScalerTrait
+            metadata:
+              name: example-appconfig-trait
+            spec:
+              replicaCount: 3
+      scopes:
+        - scopeRef:
+            apiVersion: core.oam.dev/v1alpha2
+            kind: HealthScope
+            name: example-health-scope
```
Comment (wonderflow) [inline examples/typed-component/sample_scope_definition.yaml:52]: please add this to  https://github.com/crossplane/oam-kubernetes-runtime/blob/fb48f9249602ca7338e308d328b291d6a65ef46b/examples/typed-component/sample_application_config.yaml

**Review by wonderflow — CHANGES_REQUESTED** [review]

File: examples/typed-component/sample_scope_definition.yaml (hunk 1 above)
Comment (LeoLiuYan) [reply examples/typed-component/sample_scope_definition.yaml]: ok

File: examples/typed-component/sample_scope_definition.yaml (hunk 2 above)
Comment (LeoLiuYan) [reply examples/typed-component/sample_scope_definition.yaml]: done

**Review by wonderflow — APPROVED** [review]

File: examples/typed-component/sample_application_config.yaml [hunk 3]
```diff
@@ -2,8 +2,17 @@ apiVersion: core.oam.dev/v1alpha2
 kind: HealthScope
 metadata:
   name: example-health-scope
+  namespace: oam
```
Comment (wonderflow) [inline examples/typed-component/sample_application_config.yaml:5]: can you remove this namespace?

Comment (wonderflow) [reply examples/typed-component/sample_application_config.yaml:5]: This is not align with other object

Comment (LeoLiuYan) [reply examples/typed-component/sample_application_config.yaml]: That's fine.

**Pull request merged**